### ISSUE TITLE
code cli shouldn't panic if it can't bind a port

### ISSUE THREAD

**Issue by leifwalsh** [microsoft/vscode#192229]
```
❯ ./code-insiders serve-web
*
* Visual Studio Code Server
*
* By using the software, you agree to
* the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
* the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
*
Web UI available at http://127.0.0.1:8000?tkn=a1c7a8b1-bfc2-4b91-a60e-e18fac626dad
thread 'main' panicked at 'error binding to 127.0.0.1:8000: error creating server listener: Address in use (os error 98)', /home/cloudtest/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyper-0.14.26/src/server/server.rs:79:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Probably should print an error message instead of panicking, and maybe do that before printing the URL to connect to.

**Comment by roblourens** [comment]
If I use port 8000 in another process, then run `code-insiders serve-web`, it still says `Web UI available at http://127.0.0.1:8000?tkn=...` but it's not actually available on that port. I would expect it to pick 8001 or something.

**Comment by connor4312** [comment]
Make sure you're binding to the same interface. serve-web always binds to 127.0.0.1. Depending on your OS and network, that may not conflict with a service listening on `localhost` if localhost is ipv6.
